common/optional/types.go

```diff
@@ -22,59 +22,71 @@ import (
 	common "github.com/ChainSafe/gossamer/common"
 )
 
+// Uint32 struct
```

Comment:
Uint32 represents an optional uint32 type

Reply:
will update this and try to provide better description to others.